### PR TITLE
OCPQE-13468: Add Quay OCI Conformance Testing into Prow

### PR DESCRIPTION
**Pull request by xiaozhisean**
1, Run Conformance test from https://github.com/opencontainers/distribution-spec/tree/main/conformance 
2, Call step`quay-tests-enable-quay-catalogsource` to install from index image build
3, Archive result into ARTIFACT_DIR

image

### CODE REVIEW

**Comment by SeanZhao-redhat** [comment]
periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse  periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

File: ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+tests:
+- as: quay-conformance-test-quay311
+  cron: 0 12 1 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      COMPUTE_NODE_TYPE: m5.8xlarge
+      ODF_OPERATOR_CHANNEL: stable-4.15
+      QUAY_OPERATOR_CHANNEL: stable-3.11
```
Comment (LiZhang19817) [inline ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml:34]: Need to add new env variable to test un-released Quay Version, like 3.12

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-ocp-416-quay-quay-e2e-tests-quay39-ocp416

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse  periodic-ci-quay-quay-tests-master-ocp-416-quay-quay-e2e-tests-quay311-ocp416-lp-interop

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-ocp-416-quay-quay-e2e-tests-quay310-ocp416

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-dast-quay-dast-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-conformance-quay-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-dast-quay-dast-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-ocp-416-quay-quay-e2e-tests-quay311-ocp416-lp-interop

File: ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml (hunk 1 above)
Comment (SeanZhao-redhat) [reply ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml:35]: Done, add env variable QUAY_INDEX_IMAGE_BUILD

File: ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    test:
+    - ref: quay-tests-deploy-quay-odf
+    - ref: quay-tests-quay-conformance-test
+    workflow: cucushift-installer-rehearse-aws-ipi
+zz_generated_metadata:
+  branch: master
+  org: quay
+  repo: quay-tests
+  variant: quay-conformance
```
Comment (LiZhang19817) [inline ci-operator/config/quay/quay-tests/quay-quay-tests-master__quay-conformance.yaml:44]: suggest use name "quay-oci-conformance" to show that's **OCI Conformance Testing**

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse  periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

File: ci-operator/step-registry/quay-tests/quay-oci-conformance-test/quay-tests-quay-oci-conformance-test-commands.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Extra settings
+export OCI_HIDE_SKIPPED_WORKFLOWS=0
+export OCI_DEBUG=0
+export OCI_DELETE_MANIFEST_BEFORE_BLOBS=0 # defaults to OCI_DELETE_MANIFEST_BEFORE_BLOBS=1 if not set
+
+#Generate test result to ARTIFACT_DIR
+export OCI_REPORT_DIR
+
+echo "Begin conformance test"
+./conformance.test
```
Comment (LiZhang19817) [inline ci-operator/step-registry/quay-tests/quay-oci-conformance-test/quay-tests-quay-oci-conformance-test-commands.sh:41]: As this Prow CI Job has triggered Quay OCI Conformance testing successfully, and most of the testing case are passed, suggest not failing the whole CI job with "./conformance.test || true"

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

**Comment by SeanZhao-redhat** [comment]
/pj-rehearse periodic-ci-quay-quay-tests-master-quay-oci-conformance-quay-oci-conformance-test-quay311

File: ci-operator/step-registry/quay-tests/quay-oci-conformance-test/quay-tests-quay-oci-conformance-test-commands.sh (hunk 3 above)
Comment (SeanZhao-redhat) [reply ci-operator/step-registry/quay-tests/quay-oci-conformance-test/quay-tests-quay-oci-conformance-test-commands.sh]: updated

**Pull request merged**